bootstrap.sh

```diff
@@ -270,9 +270,29 @@ if [ -d /etc/appscale/certs ]; then
                 echo "e.g.: git stash; git checkout <appscale-tools-version>"
                 exit 1
             fi
-        else
+        elif [ FORCE_UPGRADE = "N" ]; then
             (cd appscale; git pull)
             (cd appscale-tools; git pull)
+        else
+            RANDOM_KEY="$(echo $(date), $$|md5sum|head -c 6)-$(date +%s)"
+            if ! (cd appscale;
+                    git remote add -f appscale-bootstrap-"${RANDOM_KEY}" "${APPSCALE_REPO}";
+                    git checkout appscale-bootstrap-"${RANDOM_KEY}"/"${APPSCALE_BRANCH}"); then
+                echo "Please stash your local unsaved changes and checkout"\
```

Comment:
that would do. Or 'please make sure you are on the unmodified branch (git will refuse to pull)'